xds/internal/xdsclient/xds.go

```diff
@@ -323,8 +323,35 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) ([]HTTPFilter, error)
 			}
 			if !seenHCM {
 				// TODO: Implement terminal filter logic, as per A36.
-				httpFilters = filters
+				filterChain.HTTPFilters = filters
 				seenHCM = true
+				switch hcm.RouteSpecifier.(type) {
```

Comment:
Yeah, the same. Both client and server side for route configuration can specify either an inline or rds, as you know. Should I try and find a way to combine them? In #4610 of course :P?

Reply:
Make it a function that returns `(string, *RouteConfigUpdate, error)`

Reply:
Sg, will do. Will wait for any comments @easwars has so I can do everything atomically.

Reply:
Actually, Easwar made note of this line from Eric's spec: "If an XdsServer implementation uses RouteConfiguration or supports any HTTP filters other than the hard-coded Router, then `HttpConnectionManager.route_config` and `HttpConnectionManager.rds` must be supported and RouteConfigurations must be validated. RouteConfiguration". Thus, this line is incorrect, as unlike the client side, the server side doesn't require the RouteSpecifier to be present. Since it will not share the same logic, I will keep it in this function (which will be moved to filter_chain.go).